intake/catalog/zarr.py

```diff
@@ -0,0 +1,58 @@
+from .base import Catalog
+from .local import LocalCatalogEntry
+from ..source import register_driver
+
+
+class ZarrCatalog(Catalog):
+    """Catalog as described by a Zarr group."""
+    version = '0.0.1'
+    container = 'catalog'
+    partition_access = None
+    name = 'zarr_cat'
+
+    def __init__(self, url, consolidated=False, **kwargs):
+        self._url = url
+        self._consolidated = consolidated
+        self._grp = None
+        super().__init__(**kwargs)
+
+    def _load(self):
+        import zarr
+
+        if self._grp is None:
+            # obtain the zarr group to wrap as a catalog
+            if isinstance(self._url, zarr.hierarchy.Group):
+                # use already-opened group
+                self._grp = self._url
+            else:
+                # open root group from url
+                from fsspec import get_mapper
+                store = get_mapper(self._url)
+                if self._consolidated:
+                    # use consolidated metadata
+                    self._grp = zarr.open_consolidated(store=store, mode='r')
+                else:
+                    self._grp = zarr.open_group(store=store, mode='r')
+
+        # build catalog entries
+        entries = {}
+        for k, v in self._grp.items():
+            if isinstance(v, zarr.core.Array):
+                entry = LocalCatalogEntry(name=k,
+                                          description='',
+                                          driver='ndzarr',
+                                          args=dict(url=v),
```

Comment:
We end up making a new mapper for each child that gets accessed, which may be slow for remote storage. It would be possible to pass on the group/array on the already-instantiated store, particularly for the group case in the next branch.

Reply:
Here the actual child zarr array or group (`v`) is passed through as the value of the `url` argument, which does work and means that the mapper is only used once at the root (I think!)